controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -545,3 +553,49 @@ func firstNodeNotMatchingName(name string, nodes []corev1.Node) *corev1.Node {
 	}
 	return nil
 }
+
+// modifyImageTag takes an image string, and returns an updated tagged image
+func modifyImageTag(image, tagName string) (string, error) {
+	namedRef, err := reference.ParseNormalizedNamed(image)
+	// return error if images use digest as version instead of tag
+	if _, isCanonical := namedRef.(reference.Canonical); isCanonical {
+		return "", errors.New("image uses digest as version, cannot update tag ")
+	}
+
+	// update the image tag with tagName
+	namedTagged, err := reference.WithTag(namedRef, tagName)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to update image tag")
+
+	}
+
+	// if tagging is successful, return the full image name (registry/repo:tag)
+	if _, ok := namedRef.(reference.Tagged); ok {
```

Comment:
Do you need this check? I assume if we get no error from the `reference.WithTag()` call on line 566, we should be able to return `reference.FamiliarString` directly, right?

Reply:
Correct, removed.